src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityProtocolFactory.cs

```diff
@@ -0,0 +1,1085 @@
+            //    }
+
+            //    // If DetectReplays is true and nonceCache is null then use the default InMemoryNonceCache. 
+            //    if (this.nonceCache == null)
+            //    {
+            //        // The nonce needs to be cached for replayWindow + 2*clockSkew to eliminate replays
+            //        this.nonceCache = new InMemoryNonceCache(this.ReplayWindow + this.MaxClockSkew + this.MaxClockSkew, this.MaxCachedNonces);
+            //    }
+            //}
+
+            //this.derivedKeyTokenAuthenticator = new NonValidatingSecurityTokenAuthenticator<DerivedKeySecurityToken>();
+            return Task.CompletedTask;
+        }
+
+        public virtual Task OnCloseAsync(TimeSpan timeout)
+        {
+            this.OnClose(timeout);
+            return Task.CompletedTask;
+        }
+
+       
+        internal void Open(string propertyName, bool requiredForForwardDirection, SecurityTokenAuthenticator authenticator, TimeSpan timeout)
+        {
+            if (authenticator != null)
+            {
+                TimeoutHelper helper = new TimeoutHelper(timeout);
+                SecurityUtils.OpenTokenAuthenticatorIfRequiredAsync(authenticator, helper.GetCancellationToken());
+            }
+            else
+            {
+                OnPropertySettingsError(propertyName, requiredForForwardDirection);
+            }
+        }
+
+        internal void Open(string propertyName, bool requiredForForwardDirection, SecurityTokenProvider provider, TimeSpan timeout)
+        {
+            if (provider != null)
+            {
+                SecurityUtils.OpenTokenProviderIfRequiredAsync(provider,  new TimeoutHelper(timeout).GetCancellationToken());
+            }
+            else
+            {
+                OnPropertySettingsError(propertyName, requiredForForwardDirection);
+            }
+        }
+
+        internal void OnPropertySettingsError(string propertyName, bool requiredForForwardDirection)
+        {
+            if (requiredForForwardDirection)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(
+                    SR.Format(SR.PropertySettingErrorOnProtocolFactory, propertyName, this),
+                    propertyName));
+            }
+            else if (this.requestReplyErrorPropertyName == null)
+            {
+                this.requestReplyErrorPropertyName = propertyName;
+            }
+        }
+
+        private void ThrowIfReturnDirectionSecurityNotSupported()
+        {
+            if (this.requestReplyErrorPropertyName != null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(
+                    SR.Format(SR.PropertySettingErrorOnProtocolFactory, this.requestReplyErrorPropertyName, this),
+                    this.requestReplyErrorPropertyName));
+            }
+        }
+
+        internal void ThrowIfImmutable()
```

Comment:
All these methods need to be implemented and not throwing as they get called when Open is called via WrapperSecurityCommunicationObject. I'd like to understand why things aren't breaking right now with OnOpened throwing. That suggests we aren't calling Open on this correctly somewhere.

Reply:
these are enabled now... you are right, opening wasn't happening properly after removing comment, as there were bugs... fixed all of them....